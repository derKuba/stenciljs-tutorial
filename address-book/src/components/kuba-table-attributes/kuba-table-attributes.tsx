import { Component, Host, h, Prop } from '@stencil/core';

@Component({
  tag: 'kuba-table-attributes',
  styleUrl: 'kuba-table-attributes.css',
  shadow: true,
})
export class KubaTableAttributes {

  @Prop() head;
  @Prop() body;

  render() {
    return (
      <Host>
        <table>
          <thead>
            <tr>
              {
                this.head.length > 0 && this.head.map(row => <th>{row}</th>)
              }
            </tr>
          </thead>
          <tbody>
            {
              this.body.length > 0 && this.body.map(row => <tr>
                {row.map(cell => <td>{cell}</td>)}
              </tr>)
            }
          </tbody>
        </table>
      </Host>
    );
  }
}
