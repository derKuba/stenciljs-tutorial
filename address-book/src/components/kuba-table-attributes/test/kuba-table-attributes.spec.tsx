import { newSpecPage } from "@stencil/core/testing";
import { h } from "@stencil/core";
import { KubaTableAttributes } from "../kuba-table-attributes";

describe("kuba-table-attributes", () => {
  it("renders", async () => {
    const head = ["id", "Name", "Vorname"];
    const body = [
      ["1", "Max", "Muster"],
      ["2", "Maxine", "Muster"],
      ["3", "Momo", "Muster"],
    ];

    const page = await newSpecPage({
      components: [KubaTableAttributes],
      template: () => <kuba-table-attributes head={head} body={body} />,
    });
    expect(page.root).toMatchSnapshot();
  });
});
