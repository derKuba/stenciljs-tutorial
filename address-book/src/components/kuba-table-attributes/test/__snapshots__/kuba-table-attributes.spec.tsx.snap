// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kuba-table-attributes renders 1`] = `
<kuba-table-attributes>
  <mock:shadow-root>
    <table>
      <thead>
        <tr>
          <th>
            id
          </th>
          <th>
            Name
          </th>
          <th>
            Vorname
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            Max
          </td>
          <td>
            Muster
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td>
            Maxine
          </td>
          <td>
            Muster
          </td>
        </tr>
        <tr>
          <td>
            3
          </td>
          <td>
            Momo
          </td>
          <td>
            Muster
          </td>
        </tr>
      </tbody>
    </table>
  </mock:shadow-root>
</kuba-table-attributes>
`;
